### PR TITLE
[clang-tidy] modernize-use-emplace

### PR DESCRIPTION
**Pull request by Rechi**
## Description
apply and enable `modernize-use-emplace` clang-tidy check

## How has this been tested?
run clang-tidy and compiled for Android (aarch64), iOS, Linux (gl and gles), macOS and tvOS

## Types of change
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Review by garbear — APPROVED** [review]
Game API change looks good, and I checked the rest of the changes briefly and saw no issues.

**Comment by ksooo** [comment]
> How has this been tested?
> run clang-tidy and compiled

No runtime test? We have seen regressions more than one time which were introduced by this kind of PRs. I recently discovered one that sneaked in in February this year and spent literally a whole weekend to find the issue. Now being in beta for Omega I'm a bit hesitant to give my approval for this PR, tbh.

**Comment by garbear** [comment]
I run-time tested this PR alongside https://github.com/xbmc/xbmc/pull/23987. I was mostly testing games and everything looks good there. I also tried playing a few videos and everything seems fine.

Given that we're still early in the beta cycle, more eyes are showing up, and we're still a ways from RC, I think this can be merged.

**Comment by garbear** [comment]
I've also included this PR in my 2023-10-24 RetroPlayer test builds when I rebased on Beta 1: https://github.com/garbear/xbmc/releases

The test build is starting to get picked up by forum users, which should give us a wider test base for this PR.

File: xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp [hunk 1]
```diff
@@ -974,8 +974,7 @@ void CGUIDialogPVRTimerSettings::ChannelsFiller(const SettingConstPtr& setting,
             !pThis->m_timerType->SupportsAnyChannel())
           continue;
 
-        list.emplace_back(
-            IntegerSettingOption(channelEntry.second.description, channelEntry.first));
+        list.emplace_back(channelEntry.second.description, channelEntry.first);
```
Comment (ksooo) [inline xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp:977]: Please revert this, it is not given that `IntegerSettingOption` will actually stay a `std::pair` forever.

File: xbmc/pvr/PVREventLogJob.cpp [hunk 2]
```diff
@@ -31,7 +31,7 @@ void CPVREventLogJob::AddEvent(bool bNotifyUser,
                                const std::string& msg,
                                const std::string& icon)
 {
-  m_events.emplace_back(Event(bNotifyUser, eLevel, label, msg, icon));
+  m_events.emplace_back(bNotifyUser, eLevel, label, msg, icon);
```
Comment (ksooo) [inline xbmc/pvr/PVREventLogJob.cpp:34]: Not sure this change is correct. Is really the right method called here still?

File: xbmc/profiles/dialogs/GUIDialogProfileSettings.cpp [hunk 3]
```diff
@@ -347,11 +347,11 @@ void CGUIDialogProfileSettings::InitializeSettings()
...
     TranslatableIntegerSettingOptions entries;
-    entries.push_back(TranslatableIntegerSettingOption(20062, 0));
-    entries.push_back(TranslatableIntegerSettingOption(20063, 1));
-    entries.push_back(TranslatableIntegerSettingOption(20061, 2));
+    entries.emplace_back(20062, 0);
+    entries.emplace_back(20063, 1);
+    entries.emplace_back(20061, 2);
     if (profileManager->GetMasterProfile().getLockMode() != LOCK_MODE_EVERYONE)
-      entries.push_back(TranslatableIntegerSettingOption(20107, 3));
+      entries.emplace_back(20107, 3);
```
Comment (ksooo) [inline xbmc/profiles/dialogs/GUIDialogProfileSettings.cpp:354]: Please revert the removal of the ctor call. It is not given that `TranslatableIntegerSettingOption` will actually stay a `std::pair` forever.

File: xbmc/pvr/epg/EpgContainer.cpp [hunk 4]
```diff
@@ -892,13 +892,13 @@ void CPVREpgContainer::SetHasPendingUpdates(bool bHasPendingUpdates /* = true */
 void CPVREpgContainer::UpdateRequest(int iClientID, int iUniqueChannelID)
 {
   std::unique_lock<CCriticalSection> lock(m_updateRequestsLock);
-  m_updateRequests.emplace_back(CEpgUpdateRequest(iClientID, iUniqueChannelID));
+  m_updateRequests.emplace_back(iClientID, iUniqueChannelID);
```
Comment (ksooo) [inline xbmc/pvr/epg/EpgContainer.cpp:895]: Not sure this change is correct. Is really the right method called here still? `CEpgUpdateRequest` is not a `std::pair`, as far as I remember. And even though it is, I would like to stick with the real type name here.

File: xbmc/pvr/epg/EpgContainer.cpp [hunk 5]
```diff
@@ -892,13 +892,13 @@ void CPVREpgContainer::SetHasPendingUpdates(bool bHasPendingUpdates /* = true */
...
 }
 
 void CPVREpgContainer::UpdateFromClient(const std::shared_ptr<CPVREpgInfoTag>& tag, EPG_EVENT_STATE eNewState)
 {
   std::unique_lock<CCriticalSection> lock(m_epgTagChangesLock);
-  m_epgTagChanges.emplace_back(CEpgTagStateChange(tag, eNewState));
+  m_epgTagChanges.emplace_back(tag, eNewState);
```
Comment (ksooo) [inline xbmc/pvr/epg/EpgContainer.cpp:901]: Not sure this change is correct. Is really the right method called here still? `CEpgTagChange` is not a `std::pair`, as far as I remember. And even though it is, I would like to stick with the real type name here.

**Review by ksooo — APPROVED** [review]
PVR changes look good.

**Comment by garbear** [comment]
jenkins build this please

**Comment by howie-f** [comment]
@Rechi i do remember some insert/emplace "confusion" while working on AddonRepos. feel free to pick this commit https://github.com/howie-f/xbmc/commit/ccaf12babfbfb4cb254d8214a53bf1ffb60f75f1

**Comment by garbear** [comment]
@Rechi Do you plan to cherry-pick? If not I'll merge and we can open @howie-f 's commit as a separate PR.

**Comment by Rechi** [comment]
As the changes aren't reported by clang-tidy I won't pick those.

**Pull request merged**